GitExtUtils/FileLogging/LogWriter.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Diagnostics.SymbolStore;
+using System.IO;
+using System.Text;
+using System.Threading;
+
+namespace GitExtUtils.FileLogging
+{
+    public interface ILogWriter
+    {
+        void Enqueue(LogEntry entry);
+    }
+
+    public class LogWriter : ILogWriter
+    {
+        private readonly LogPathGenerator _logPathGenerator;
+        private readonly ILogFormatter _formatter;
+        private readonly TimeSpan _flushInterval;
+        private readonly ConcurrentQueue<LogEntry> _unsaved = new ConcurrentQueue<LogEntry>();
+
+        public LogWriter(
+            LogPathGenerator logPathGenerator,
+            ILogFormatter formatter,
+            TimeSpan flushInterval)
+        {
+            _logPathGenerator = logPathGenerator;
+            _formatter = formatter;
+            _flushInterval = flushInterval;
+        }
+
+        public void Enqueue(LogEntry entry)
+        {
+            _unsaved.Enqueue(entry);
+        }
+
+        public void Run(CancellationToken cancellationToken)
```

Comment:
The interface declares only a single method.
This means that either this method isn't supposed to be `public` or callsites are using concrete implementations instead of the interface. 
Either way we have a problem.

Reply:
Agree we need to add it to the interface